File: services/blogApi.ts
export interface PostPayload {
  title: string;
  excerpt: string;
  content: string;
  author: string;
  cover_image: string;
  tags: string[];
  published: boolean;
}

export interface PostResponse {
  post: {
    id: string;
    slug: string;
    title: string;
    created_at: string;
  };
}

export interface PostDetail {
  id: string;
  slug: string;
  title: string;
  excerpt: string;
  content: string;
  cover_image: string;
  author: string;
  tags: string[];
  published: boolean;
  created_at: string;
}

/**
 * Fetch a single post by ID from the Jesse A. Eisenbalm blog API.
 */
export async function getPost(id: string): Promise<PostDetail> {
  const apiKey = process.env.BLOG_API_KEY;
  const apiUrl = process.env.BLOG_API_URL ?? "https://jesse-eisenbalm-server.vercel.app";

  if (!apiKey) throw new Error("BLOG_API_KEY environment variable is not set");

  const response = await fetch(`${apiUrl}/api/posts/${id}`, {
    headers: { "x-api-key": apiKey },
    cache: "no-store",
  });

  if (!response.ok) {
    const text = await response.text().catch(() => "");
    throw new Error(`Blog API error: ${response.status} ${response.statusText} — ${text}`);
  }

  const data = (await response.json()) as { post: PostDetail };
  return data.post;
}

/**
 * Create a post on the Jesse A. Eisenbalm blog API.
 * Set published: true to go live, false to save as draft.
 */
export async function createPost(payload: PostPayload): Promise<PostResponse> {
  const apiKey = process.env.BLOG_API_KEY;
  const apiUrl = process.env.BLOG_API_URL ?? "https://jesse-eisenbalm-server.vercel.app";

  if (!apiKey) {
    throw new Error("BLOG_API_KEY environment variable is not set");
  }

  const response = await fetch(`${apiUrl}/api/posts`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "x-api-key": apiKey,
    },
    body: JSON.stringify(payload),
  });

  if (!response.ok) {
    const text = await response.text().catch(() => "");
    throw new Error(
      `Blog API error: ${response.status} ${response.statusText} — ${text}`
    );
  }

  const data = (await response.json()) as PostResponse;
  if (!data.post?.id) {
    throw new Error("Blog API response missing post.id");
  }

  return data;
}

File: backend/services/blog_api.py
"""Create posts on the Jesse A. Eisenbalm blog API."""
from __future__ import annotations

import os
from dataclasses import dataclass

import httpx


@dataclass
class PostResponse:
    id: str
    slug: str
    title: str
    created_at: str


def create_post(
    title: str,
    excerpt: str,
    content: str,
    author: str,
    cover_image: str,
    tags: list[str],
    published: bool,
) -> PostResponse:
    api_key = os.environ["BLOG_API_KEY"]
    api_url = os.environ.get("BLOG_API_URL", "https://jesse-eisenbalm-server.vercel.app")

    response = httpx.post(
        f"{api_url}/api/posts",
        json={
            "title": title,
            "excerpt": excerpt,
            "content": content,
            "author": author,
            "cover_image": cover_image,
            "tags": tags,
            "published": published,
        },
        headers={
            "Content-Type": "application/json",
            "x-api-key": api_key,
        },
        timeout=30.0,
    )

    if not response.is_success:
        raise RuntimeError(
            f"Blog API error: {response.status_code} {response.reason_phrase} — {response.text[:300]}"
        )

    data = response.json()
    post = data.get("post") or {}
    if not post.get("id"):
        raise RuntimeError("Blog API response missing post.id")

    return PostResponse(
        id=post["id"],
        slug=post.get("slug", ""),
        title=post.get("title", ""),
        created_at=post.get("created_at", ""),
    )
